ext/js/data/sandbox/anki-note-data-creator.js

```diff
@@ -340,6 +344,68 @@ function getKanjiFrequencies(dictionaryEntry) {
     return results;
 }
 
+/**
+ * @param {import('dictionary').KanjiDictionaryEntry} dictionaryEntry
+ * @returns {number}
+ */
+function getKanjiFrequencyHarmonic(dictionaryEntry) {
+    let total = 0;
+    let length = 0;
+    let previousDictionary;
+
+    if (dictionaryEntry.frequencies.length < 1) {
+        return -1;
+    }
+
+    for (const {dictionary, frequency, displayValue} of dictionaryEntry.frequencies) {
+        if (dictionary === previousDictionary) {
+            continue;
+        }
+        previousDictionary = dictionary;
+        length += 1;
+
+        const frequencyWorking = displayValue !== null ? displayValue.match(/\d+/) : frequency;
+        if (frequencyWorking !== null) {
+            total += 1 / Number(frequencyWorking);
+        } else {
+            total += 1 / frequency;
+        }
+    }
+    total = Math.floor(length / total);
+    return total;
+}
+
+/**
+ * @param {import('dictionary').KanjiDictionaryEntry} dictionaryEntry
+ * @returns {number}
+ */
+function getKanjiFrequencyAverage(dictionaryEntry) {
+    let total = 0;
+    let length = 0;
+    let previousDictionary;
+
+    if (dictionaryEntry.frequencies.length < 1) {
+        return -1;
+    }
+
+    for (const {dictionary, frequency, displayValue} of dictionaryEntry.frequencies) {
+        if (dictionary === previousDictionary) {
+            continue;
+        }
+        previousDictionary = dictionary;
+        length += 1;
+
+        const frequencyWorking = displayValue !== null ? displayValue.match(/\d+/) : frequency;
+        if (frequencyWorking !== null) {
+            total += Number(frequencyWorking);
```

Comment:
Repeated code between this and the last one, make a helper method to get the computed integer value of a frequency.

```js
function getFrequencyNumber(value, displayValue) {
  if (displayValue !== null) {
    const match = displayValue.match(/\d+/);
    if (match !== null) { return Number.parseInt(match[0], 10); }
  }
  return value;
}
```